pulsar-client/src/test/java/org/apache/pulsar/client/impl/AutoClusterFailoverTest.java

```diff
@@ -283,4 +291,46 @@ public void testAutoClusterFailoverSwitchTlsTrustStore() throws IOException {
                 .updateTlsTrustStorePathAndPassword(primaryTlsTrustStorePath, primaryTlsTrustStorePassword);
 
     }
+
+    @Test
+    public void testUrl() throws Exception {
+        Random random = new Random(42);
+        String serviceUrl = "pulsar://localhost:" + random.nextInt(65535);
+
+
+        PulsarServiceNameResolver resolver = new PulsarServiceNameResolver();
+
+        try {
+            resolver.updateServiceUrl(serviceUrl);
+            InetSocketAddress inetSocketAddress = new InetSocketAddress(resolver.resolveHost().getHostName(), resolver.resolveHost().getPort());
+            Socket socket = new Socket();
+            socket.connect(inetSocketAddress, 30);
+            socket.close();
+            fail();
```

Comment:
Seems this could be flaky? If the random port is open by coincidence.

Reply:
it is better to use a ephemeral port
btw I am not convinced if it is worth to add such test.

I believe that we can drop the test at all

Reply:
> Seems this could be flaky? If the random port is open by coincidence.

+1